src/App.tsx

```diff
@@ -1,16 +1,87 @@
 import React from 'react';
 import './App.scss';
 
-class App extends React.Component {
-  addOne = () => {};
+type State = {
+  count: number;
+};
 
-  add100 = () => {};
+class App extends React.Component<{}, State> {
+  state = {
+    count: 0,
+  };
 
-  increase = () => {};
+  addOne = () => {
+    this.setState((state) => ({
+      count: state.count + 1,
+    }));
+  };
+
+  add100 = () => {
+    this.setState((state) => ({
+      count: state.count + 100,
+    }));
+  };
+
+  increase = () => {
+    if (this.state.count % 5 === 0) {
+      this.setState(() => {
+        return this.add100();
+      });
```

Comment:
```suggestion
        this.add100();
```
В this.add100() вже і так є метод this.setState для чого 2 раза його визивати ?